src/ImageSharp/Formats/Webp/WebpDecoderCore.cs

```diff
@@ -456,7 +462,12 @@ private void ParseOptionalExtendedChunks(WebpChunkType chunkType, WebpFeatures f
                     else
                     {
                         byte[] exifData = new byte[exifChunkSize];
-                        this.currentStream.Read(exifData, 0, (int)exifChunkSize);
+                        bytesRead = this.currentStream.Read(exifData, 0, (int)exifChunkSize);
+                        if (bytesRead != exifChunkSize)
+                        {
+                            WebpThrowHelper.ThrowInvalidImageContentException("Not enough data to read the exif chunk");
```

Comment:
Are there rules in WebP about where a these chunks should be. I.E are they always before the image data?

Reply:
ICCP and ANIM chunks should always come before the image data, EXIF and XMP chunks should come after the image data.

https://developers.google.com/speed/webp/docs/riff_container#extended_file_format

Reply:
If after, I don't think we should throw then and skip instead. I would consider those non-critical.